src/DynamoCore/Graph/Workspaces/UndoRedo.cs

```diff
@@ -34,15 +34,35 @@ internal UndoRedoRecorder UndoRecorder
         /// </summary>
         public bool CanUndo
         {
-            get { return ((null != undoRecorder) && undoRecorder.CanUndo); }
+            get
+            {
+                if (UndoRedoLocked)
+                {
+                    return false;
+                }
+                return ((null != undoRecorder) && undoRecorder.CanUndo);
```

Comment:
Maybe simply as `return ((null != undoRecorder) && undoRecorder.CanUndo) && !IsUndoRedoLocked;` so that you dont have to nest the condition above